NavigationBasicSample/app/src/androidTest/java/com/example/pavel/GameOverDestinationTest.kt

```diff
@@ -0,0 +1,25 @@
+import android.support.test.espresso.Espresso.onView
+import android.support.test.espresso.action.ViewActions.click
+import android.support.test.espresso.matcher.ViewMatchers
+import android.support.test.espresso.matcher.ViewMatchers.withId
+import com.example.android.navigationsample.R
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class GameOverDestinationTest : BaseNavigationTest() {
+
+    override fun navigateToTestStartDestination() {
+        onView(withId(R.id.play_btn)).perform(click())
+        onView(withId(R.id.signup_btn)).perform(click())
+        onView(withId(R.id.play_btn3)).perform(click())
+        onView(withId(R.id.checkBox)).perform(click())
+    }
+
+    @Test
+    fun testGoingToMatchDestination() {
+        navigator.addOnNavigatorNavigatedListener { _, destId, _ ->
+            assertEquals(destId, R.id.match)
+        }
+        onView(ViewMatchers.withId(R.id.play_btn4)).perform(click())
```

Comment:
Consider using a [countdown latch](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/CountDownLatch.html) to wait till navigation listener is triggered. As shown below. 
```
val latch = CountDownLatch(1)
navigator.addOnNavigatorNavigatedListener { _, destId, _ ->
            assertEquals(destId, R.id.match)
            latch.countDown()
        }
        onView(ViewMatchers.withId(R.id.play_btn4)).perform(click())
        latch.await(1, TimeUnit.SECONDS)
```